docs/source/release_notes.rst

```diff
@@ -6,6 +6,7 @@ Release Notes
         * ``IDColumnsDataCheck`` now only returns an action code to set the first column as the primary key if it contains unique values :pr:`3639`
         * ``IDColumnsDataCheck`` now can handle primary key columns containing "integer" values that are of the double type :pr:`3683` 
         * Added support for BooleanNullable columns in EvalML pipelines and imputer :pr:`3678`
+        * Only apply ``StandardScaler`` to numeric columns :pr:`3686`
```

Comment:
Mega tiny itsy bitsy nit: `Only apply StandardScaler to numeric columns` -> `Updated StandardScaler to only apply to numeric columns`

I'm way too pedantic about the past tense 😁